File: Leetcode/Leetcode DSA course/Arrays and Strings/Sliding window Technique/2-Max sum subarray of size k/main.java
public class Solution{
    public int getMaxSum(int [] nums, int k){
        int curr=0;
        for(int i=0;i<k;i++)
            curr+=nums[i];
        int ans=curr;
        for(int i=k;i<nums.length;i++){
            curr+=nums[i] - nums[i-k];
            ans = Math.max(ans,curr);
        }

        return ans;
        
    }
}

File: Leetcode/Leetcode DSA course/Arrays and Strings/Sliding window Technique/2-Max sum subarray of size k/main.py

from typing import List
class Solution:
    def getMaxSum(self, nums:List[int], k:int) -> int:
        curr=0
        for i in range(k):
            curr+=nums[i]
        ans=curr
        for i in range(k,len(nums)):
            curr+=nums[i]-nums[i-k]
            ans = max(ans, curr)
        return ans

nums = [3,-1,4,12 ,-8,5,6]
k=4
solution = Solution()
print(solution.getMaxSum(nums,k))

